### PR TITLE
MM-45509 - Disable and implement upgrade flow on RTCD Service URL config option

### PR DESCRIPTION
**Pull request by cpoile**
#### Summary
- Improved how we do i18n -- we don't need to specify an id, it will be generated automatically, and strings in `en.json` will be updated automatically if the default message is changed in the code.
- Upgrade flow is done, needs proper links and copy edit @justinegeffen 

#### Ticket Link
- https://mattermost.atlassian.net/browse/MM-45509

#### Screenshots
image
image
image
image

### CODE REVIEW

File: webapp/src/components/admin_console_settings/rtcd_service_url/index.tsx [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+                            />
+                        </Title>
+                        <VerticalSpacer size={8}/>
+                        <Text>
+                            <FormattedMessage
+                                defaultMessage={'Real-time communication daemon is a service built to offload calls on your own WebRTC services in order to efficiently support scalable and secure deployments. <featureLink>Learn more about this feature</featureLink>.'}
+                                values={{
+                                    featureLink: (text: string) => (
+                                        <a
+                                            href='TODO'
```
Comment (cpoile) [inline webapp/src/components/admin_console_settings/rtcd_service_url/index.tsx:91]: This link is needed

File: webapp/src/components/admin_console_settings/rtcd_service_url/modals.tsx [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+                        '<guestAccountsLink>guest accounts</guestAccountsLink>, ' +
+                        '<autoComplianceReportsLink>automated compliance reports</autoComplianceReportsLink>, and ' +
+                        '<mobileSecureNotificationsLink>mobile secure-ID push notifications</mobileSecureNotificationsLink>, ' +
+                        'among many other features. ' +
+                        'View all features in our <documentationLink>documentation</documentationLink>.'}
+                    values={{
+                        trialExpirationDate: trialEndDate,
+                        rtcdDocsLink: (text: string) => (
+                            <a
+                                href='TODO'
```
Comment (cpoile) [inline webapp/src/components/admin_console_settings/rtcd_service_url/modals.tsx:61]: And this

File: webapp/src/components/admin_console_settings/rtcd_service_url/modals.tsx [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+                        mobileSecureNotificationsLink: (text: string) => (
+                            <NavLink
+                                to='/admin_console/environment/push_notification_server'
+                            >
+                                {text}
+                            </NavLink>
+                        ),
+                        documentationLink: (text: string) => (
+                            <a
+                                href='TODO'
```
Comment (cpoile) [inline webapp/src/components/admin_console_settings/rtcd_service_url/modals.tsx:91]: And this

File: webapp/i18n/en.json [hunk 4]
```diff
@@ -1,3 +1,26 @@
 {
-  "button.channel.menu": "Disable Voice"
-}
+  "/+63eJ": "Set up RTCD Services",
```
Comment (streamer45) [inline webapp/i18n/en.json:2]: This automation is great but could losing the human readable ids become a problem?

File: webapp/src/components/admin_console_settings/rtcd_service_url/index.tsx (hunk 1 above)
Comment (streamer45) [reply webapp/src/components/admin_console_settings/rtcd_service_url/index.tsx:91]: I am thinking either we link to the self-hosted documentation or directly https://github.com/mattermost/rtcd.

File: webapp/src/components/admin_console_settings/rtcd_service_url/index.tsx [hunk 5]
```diff
@@ -0,0 +1,258 @@
...
+    const cloud = useSelector(isCloud);
+    const stats = useSelector(adminStats);
+
+    const [currentRestricted, setCurrentRestricted] = useState(restricted);
+
+    const leftCol = 'col-sm-4';
+    const rightCol = 'col-sm-8';
+
+    // Webapp doesn't pass the placeholder setting.
+    const placeholder = 'https://rtcd.example.com';
```
Comment (streamer45) [inline webapp/src/components/admin_console_settings/rtcd_service_url/index.tsx:36]: I think we can import this from `src/manifest.` Also, would it make sense to update `plugin.json` and set the type of setting as `custom`?

**Review by streamer45 — COMMENTED** [review]
Great :raised_hands: , just left a couple of minor comments.

File: webapp/i18n/en.json (hunk 4 above)
Comment (streamer45) [reply webapp/i18n/en.json:2]: Discussed this internally. Same approach has been used in Playbooks and it's working well.

File: webapp/src/components/admin_console_settings/rtcd_service_url/index.tsx (hunk 5 above)
Comment (cpoile) [reply webapp/src/components/admin_console_settings/rtcd_service_url/index.tsx:36]: Good idea! Done.

File: webapp/i18n/en.json [hunk 6]
```diff
@@ -1,3 +1,26 @@
...
+  "/WMCDd": "Something went wrong!",
+  "0kSG/+": "Notify Admin",
+  "3wS4fn": "Upgrade to use calls in Channels",
+  "47FYwb": "Cancel",
+  "9tBhzB": "Upgrade now",
+  "AvpuWj": "Real-time communication daemon is a service built to offload calls on your own WebRTC services in order to efficiently support scalable and secure deployments. <featureLink>Learn more about this feature</featureLink>.",
```
Comment (justinegeffen) [inline webapp/i18n/en.json:8]:
```suggestion
  "AvpuWj": "Real-time communication daemon is a service built to offload calls onto your own WebRTC services and efficiently support scalable and secure deployments. <featureLink>Learn more about this feature</featureLink>.",
```

File: webapp/i18n/en.json [hunk 7]
```diff
@@ -1,3 +1,26 @@
...
+  "CAB3IU": "Looks like something went wrong with your Enterprise trial request. You can try again later or <supportLink>contact support</supportLink> if the error persists.",
```
Comment (justinegeffen) [inline webapp/i18n/en.json:9]:
```suggestion
  "CAB3IU": "It looks like something went wrong with your Enterprise trial request. You can try again later or <supportLink>contact Support</supportLink> if the error persists.",
```

File: webapp/i18n/en.json [hunk 8]
```diff
@@ -1,3 +1,26 @@
...
+  "CJC2pM": "Calls are a quick, audio-first way of interacting with your colleagues. Upgrade to Mattermost Professional to use calls in channels and group messages.",
```
Comment (justinegeffen) [inline webapp/i18n/en.json:10]:
```suggestion
  "CJC2pM": "Calls are a quick, audio-first, way to interact with your team. Upgrade to Mattermost Professional to use calls in channels and group messages.",
```

File: webapp/i18n/en.json [hunk 9]
```diff
@@ -1,3 +1,26 @@
...
+  "EaucLA": "View plans",
+  "H79wnj": "Welcome to your Mattermost Enterprise trial! It expires on {trialExpirationDate}. You now have access to <rtcdDocsLink>RTCD services</rtcdDocsLink>, <guestAccountsLink>guest accounts</guestAccountsLink>, <autoComplianceReportsLink>automated compliance reports</autoComplianceReportsLink>, and <mobileSecureNotificationsLink>mobile secure-ID push notifications</mobileSecureNotificationsLink>, among many other features. View all features in our <documentationLink>documentation</documentationLink>.",
+  "N2IrpM": "Confirm",
+  "TdTXXf": "Learn more",
+  "Zx9T5i": "Your trial has started!{br}Explore the benefits of Enterprise",
+  "aVENjO": "By selecting <b>Try free for 30 days</b>, I agree to the <linkEvaluation>Mattermost Software Evaluation Agreement</linkEvaluation>, <linkPrivacy>Privacy Policy</linkPrivacy>, and receiving product emails.",
+  "bvd1gK": "Try channel calls with a free trial",
+  "cLA9Ho": "Try free for 30 days",
+  "cyR7Kh": "Back",
+  "g+KEd+": "Enable Calls",
```
Comment (justinegeffen) [inline webapp/i18n/en.json:20]:
```suggestion
  "g+KEd+": "Enable calls",
```

File: webapp/i18n/en.json [hunk 10]
```diff
@@ -1,3 +1,26 @@
...
+  "gZlFBP": "Okay",
+  "hQkROI": "Use your own WebRTC service for calls and media processing",
+  "isWX8N": "Select Notify Admin to send an automatic request to your System Admins to start the trial.",
+  "maWN66": "Calls are a quick, audio-first way of interacting with your colleagues. Get the full calls experience when you start a free, 30-day trial.",
```
Comment (justinegeffen) [inline webapp/i18n/en.json:24]:
```suggestion
  "maWN66": "Calls are a quick, audio-first, way to interact with your team. Get the full calls experience when you start a free, 30-day trial.",
```

File: webapp/i18n/en.json [hunk 11]
```diff
@@ -1,3 +1,26 @@
...
+  "vWE77O": "Disable Calls"
```
Comment (justinegeffen) [inline webapp/i18n/en.json:25]:
```suggestion
  "vWE77O": "Disable calls"
```

**Review by streamer45 — APPROVED** [review]
Great work :+1:

**Comment by cpoile** [comment]
@itao Yes, correct. That's as it was in the designs, can I defer to you and @tanmay-des and I'll make whatever changes needed.

**Comment by itao** [comment]
Since code-freeze is tomorrow and I don't think this is worth blocking that, I'll approve the PR and bring the convo to channel on community.

**Review by itao — APPROVED** [review]

File: webapp/src/components/admin_console_settings/rtcd_service_url/index.tsx (hunk 1 above)
Comment (cpoile) [reply webapp/src/components/admin_console_settings/rtcd_service_url/index.tsx:92]: Sure, we could do that unless we have something more specific.  
@justinegeffen Do you have specific links that you'd like (for this and the other two below)?

File: webapp/src/components/admin_console_settings/rtcd_service_url/modals.tsx (hunk 2 above)
Comment (cpoile) [reply webapp/src/components/admin_console_settings/rtcd_service_url/modals.tsx:61]: using: https://docs.mattermost.com/configure/calls-deployment.html

File: webapp/src/components/admin_console_settings/rtcd_service_url/modals.tsx (hunk 3 above)
Comment (cpoile) [reply webapp/src/components/admin_console_settings/rtcd_service_url/modals.tsx:91]: using: https://docs.mattermost.com/configure/calls-deployment.html

File: webapp/src/components/admin_console_settings/rtcd_service_url/index.tsx (hunk 1 above)
Comment (cpoile) [reply webapp/src/components/admin_console_settings/rtcd_service_url/index.tsx:92]: using: https://docs.mattermost.com/configure/calls-deployment.html

**Pull request merged**